### PR TITLE
r.circle: Add tests for raster circle tool

### PR DESCRIPTION
**Pull request by Shreshth-Malik**
This PR adds unit tests for the `r.circle` module. The tests include:

Binary Circle Test: Verifies the creation of an output and a binary raster map using the -b flag.
Distance Circle Test: Checks the output of an output and a distance-based circle raster without the -b flag.
Multiplier flag test: Checks the output of an output after giving a multiplier as input.

### CODE REVIEW

File: raster/r.circle/testsuite/test_circle.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from grass.gunittest.gmodules import SimpleModule
+
+
+class TestRCircle(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        """Set up a temporary region for testing."""
+        cls.use_temp_region()
+        cls.runModule("g.region", n=1000, s=0, e=1000, w=0, res=10)
```
Comment (echoix) [inline raster/r.circle/testsuite/test_circle.py:13]: Is 100x100 the smallest needed to be able to test correctly this module?

Comment (Shreshth-Malik) [reply raster/r.circle/testsuite/test_circle.py]: I have reduced the region size to a smaller size. Please let me know if there is some other feedback.

**Review by petrasovaa — CHANGES_REQUESTED** [review]
Please also test the multiplier option.

File: raster/r.circle/testsuite/test_circle.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+from grass.gunittest.case import TestCase
+from grass.gunittest.main import test
+import grass.script as gs
+from grass.gunittest.gmodules import SimpleModule
+
+
+class TestRCircle(TestCase):
+
```
Comment (petrasovaa) [inline raster/r.circle/testsuite/test_circle.py:8]: You can define the output map here
```suggestion
    output = "test_circle"
```

and then refer to it in `tearDown`. You can use the same output in all of the tests.

Comment (Shreshth-Malik) [reply raster/r.circle/testsuite/test_circle.py:8]: Added the output map in `setUpClass`, it made it easier to reference later and added the same reference in `tearDown`. Thank you for the suggestion.

**Review by petrasovaa — APPROVED** [review]

**Pull request merged**